Dockerfile

```diff
@@ -0,0 +1,46 @@
+# =========================================================================
+# Stage 1: The Builder Stage
+# =========================================================================
+FROM golang:1.25-alpine AS builder
+
+WORKDIR /src
+
+# Install build dependencies like git
+RUN apk add --no-cache git
+
+# Cache dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy source and build static binaries
+COPY . .
+RUN CGO_ENABLED=0 go build -ldflags="-w -s" -o /bin/multigres ./go/cmd/multigres
+RUN CGO_ENABLED=0 go build -ldflags="-w -s" -o /bin/multiadmin ./go/cmd/multiadmin
+RUN CGO_ENABLED=0 go build -ldflags="-w -s" -o /bin/multigateway ./go/cmd/multigateway
+RUN CGO_ENABLED=0 go build -ldflags="-w -s" -o /bin/multiorch ./go/cmd/multiorch
+RUN CGO_ENABLED=0 go build -ldflags="-w -s" -o /bin/multipooler ./go/cmd/multipooler
+RUN CGO_ENABLED=0 go build -ldflags="-w -s" -o /bin/pgctld ./go/cmd/pgctld
```

Comment:
[q]: Curious, why don't we use the [make build](https://github.com/multigres/multigres/blob/5531b332fd0556e5bc975b849780081161cedf0d/Makefile#L65) here? If `build` target misses the `ldflags` and `CGO_ENABLED=0`, shouldn't we add them to the Makefile as well?